ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx

```diff
@@ -10,66 +11,71 @@ require('./workflows-toolbar.scss');
+        const actions: any = Actions.WorkflowOperationsMap;
+        const disabled = props.actionsIsDisabled;
 
-    private getNumberSelected(): number {
-        return this.props.selectedWorkflows.size;
-    }
+        return Object.keys(actions).map((actionName: WorkflowOperationName) => {
+            const action = actions[actionName];
+            return {
+                title: action.title,
+                iconClassName: action.iconClassName,
+                isDisabled: disabled[actionName],
+                action,
+                workflowsAction: async () => {
+                    // check for action
+                    const confirmed = await popup.confirm('Confirm', `Are you sure you want to ${action.title.toLowerCase()} all selected workflows?`);
+                    if (!confirmed) {
+                        return;
+                    }
 
-    private async performActionOnSelectedWorkflows(ctx: ContextApis, title: string, action: WorkflowOperationAction): Promise<any> {
-        const confirmed = await ctx.popup.confirm('Confirm', `Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
-        if (!confirmed) {
-            return Promise.resolve(false);
-        }
+                    // check for delete from archived workflows
+                    let deleteArchived = false;
+                    if (action.title === 'DELETE') {
+                        // check for delete workflows from archived workflows
+                        for (const entry of props.selectedWorkflows) {
+                            if (isArchivedWorkflow(entry[1])) {
+                                deleteArchived = await popup.confirm('Confirm', 'Do you also want to delete them from the Archived Workflows database?');
+                                break;
+                            }
+                        }
+                    }
 
-        let deleteArchived = false;
-        if (title === 'DELETE') {
-            for (const entry of this.props.selectedWorkflows) {
-                if (isArchivedWorkflow(entry[1])) {
-                    deleteArchived = await ctx.popup.confirm('Confirm', 'Do you also want to delete them from the Archived Workflows database?');
-                    break;
-                }
-            }
-        }
+                    performActionOnSelectedWorkflows(action.title, action.action, deleteArchived);
+
+                    props.clearSelection();
+                    notifications.show({
+                        content: `Performed '${action.title}' on selected workflows.`,
+                        type: NotificationType.Success
+                    });
+                },
+                disabled: () => false
+            } as WorkflowsOperation;
+        });
+    }, [props.selectedWorkflows]);
 
+    function performActionOnSelectedWorkflows(title: string, action: WorkflowOperationAction, deleteArchived: boolean): Promise<any> {
```

Comment:
```suggestion
    async function performActionOnSelectedWorkflows(title: string, action: WorkflowOperationAction, deleteArchived: boolean): Promise<any> {
```

to hopefully avoid future misses on awaiting this like the above, we could be more explicit that this is an async function (even if it only uses promise chains)